### PR TITLE
Filter duplicate IDs to ensure custom lists with attachments will always render

### PR DESCRIPTION
**Pull request by Roensby**
…ays render

### CODE REVIEW

**Comment by Roensby** [comment]
hm, looks like these tests fail for other reasons?

File: src/platform-implementation-js/dom-driver/gmail/gmail-driver/show-custom-thread-list.ts [hunk 1]
```diff
@@ -249,7 +249,16 @@ const setupSearchReplacing = (
           threadDescriptors = [];
         }
 
+        const visited = new Set<InitialPair>();
```
Comment (wegry) [inline src/platform-implementation-js/dom-driver/gmail/gmail-driver/show-custom-thread-list.ts:252]:
```suggestion
        const visited = new Set<InitialIDPair>();
```

Should patch up tests.

Comment (wegry) [reply src/platform-implementation-js/dom-driver/gmail/gmail-driver/show-custom-thread-list.ts]: You'll need to run `tsc` locally to patch up the types here.

tsc output as of [3b76776](https://github.com/InboxSDK/InboxSDK/pull/1145/commits/3b76776141fb92d13ac0559f01dff3cfa6d09ad2)

```
src/platform-implementation-js/dom-driver/gmail/gmail-driver/show-custom-thread-list.ts:255:29 - error TS2345: Argument of type 'ThreadDescriptor' is not assignable to parameter of type 'InitialIDPair'.
  Type 'string' is not assignable to type 'InitialIDPair'.

255             if (visited.has(threadDescriptor)) {
                                ~~~~~~~~~~~~~~~~

src/platform-implementation-js/dom-driver/gmail/gmail-driver/show-custom-thread-list.ts:259:25 - error TS2345: Argument of type 'ThreadDescriptor' is not assignable to parameter of type 'InitialIDPair'.

259             visited.add(threadDescriptor);
                            ~~~~~~~~~~~~~~~~


Found 2 errors in the same file, starting at: src/platform-implementation-js/dom-driver/gmail/gmail-driver/show-custom-thread-list.ts:255


Exited with code exit status 2
```

**Comment by Macil** [comment]
I've tweaked the PR so it does the duplicate check at a later stage after we've converted any message ids into thread ids which could also create more duplicates.

Thanks for investigating this. We'll get this fix out in the next version.

**Pull request merged**